### DISCUSSION TITLE
ESP32C6 zero some GPIO don´t usable as Signal input

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#21900] [Q&A] [answered]
I am currently testing the ESP32C6 zero for suitability for reading S0 counter inputs
under Tasmota 14.1.0.3 and discovered that the inputs
GPIO12, GPIO13 and GPIO18 do not work as pulse inputs.

https://de.aliexpress.com/item/1005006851575663.html?spm=a2g0o.order_list.order_list_main.5.7b785c5f1VQl7o&gatewayAdapt=glo2deu

https://www.waveshare.com/wiki/ESP32-C6-Zero

However, these GPIOs work wonderfully as relay outputs.
In my opinion, however, they should also work as pulse inputs.
Have others had this experience with ESP32C6 (zero) and what do I need to do to be able to use these inputs?

**Comment by sfromis** [comment]
On another ESP32-C6 board, I made a small test configuring gpio 18 as a Counter, wired to a push button, and it had no trouble counting the pulses as I pushed the button.

**Reply by HofeBY** [reply]
What about GPIO12 and GPIO13 ?
With external Pullup Resistor ?
I configured Chip PullUp with no external Resistor in script SML configuration, but all other GPIO are work well with this configuration.
That´s why I sucessfully testet GPIO´s as Relay Output, so I know that GPIO is OK.

**Reply by arendst** [reply, marked as answer]
Well 12 and 13 are the USB interface according to the schematics so they'll never work as you expected.

**Reply by sfromis** [reply]
Those two other pins are not broken out on my board (and not to the edge pins on your board). My use of gpio18 was straight "Counter" without modifiers, and using the internal pullup only. You could try if my test example works for you.

**Comment by HofeBY** [comment]
thx @arendst 
GPIO12/13 belogt to USB and is only usable without USB Data Connection.

**Comment by HofeBY** [comment]
thx @sfromis
I tested same condition like you.
Currently I will avoid this GPIO as Signal Input Pin.

**Reply by sfromis** [reply]
If gpio18 works for you via a button and Counter, it may then be up to how your signals behave. Maybe you need to disable the internal pullup.
